src/mempool/mempool.rs

```diff
@@ -504,6 +508,43 @@ impl Mempool {
             .message_router
             .route_fid(fid, self.read_node_mempool.num_shards);
 
+        let mut result = self
+            .insert_into_shard(shard_id, message.clone(), source.clone())
+            .await;
+
+        // Fname transfers are mirrored to both the sender and receiver shard.
+        if let MempoolMessage::ValidatorMessage(inner_message) = &message {
+            if let Some(fname_transfer) = &inner_message.fname_transfer {
+                let mirror_shard_id = self
```

Comment:
Is this the only place where the `from_fid` matters? Should we update the `fid()` function on `MempoolMessage` to return a vec of fids and return both?